wordpress-mobile/WordPress-iOS
# [Modal Layout Picker] Add analytics metrics for the layout picker

**Fixes** https://github.com/wordpress-mobile/gutenberg-mobile/issues/2454

## To test:

An easy way to test for these events is to run the app locally through XCode and filter the console to look for the string `🔵 Tracked:`

<details>
<summary><strong>To disable or enable the development version of Modal Layout Picker</strong></summary>

- Open the app from the build that allows FeatureFlags such as a PR build or a local development build
    - By default, the modal layout picker will be disabled.
    - From the site page:
        - Click on your Gravatar.
        - Click on App Settings.
        - Click on Debug.
        - Toggle "Gutenberg Modal Layout Picker" to enable or disable the picker
            <img width=300 src="https://user-images.githubusercontent.com/3384451/90816133-c8b10580-e2f9-11ea-8584-6bbfd7e523e2.gif" />
</details>

When enabled, the Layout Picker should show when creating a new page from My Site and from the Page list.

🗒️ All of these cases start by navigating to the Modal Layout Picker.
🗒️ For the `template` property mentioned below; you can expect to see values along the lines of just the category name or category name `-` number examples: `about-4` and `about` are the two layouts in the About section, `contact-6` is the layout in the Contact section. The Home Page section will have layouts like `twenty-twenty`.

### Template Create Page
1. Select a layout
1. Select Create Page
**Expect** to see the event `wpios_editor_session_template_apply` with the property `template` matching the selected layout. 
In the console, you'll see an event like: 
`🔵 Tracked: editor_session_template_apply <template: contact-6>`

### Template Preview 
1. Select a layout
1. Select Preview
**Expect** to see the event `wpios_editor_session_template_preview` with the property `template` matching the selected layout. 
In the console, you'll see an event like: 
`🔵 Tracked: editor_session_template_preview <template: contact-6>`

### Template Preview, then Create
1. Select a layout
1. Select Preview
**Expect** to see the event `wpios_editor_session_template_preview` as mentioned in the previous step
3. 🔁 Repeat selecting back and choosing a new layout as many times as you want 
3. Select Create Page
**Expect** to see the event `wpios_editor_session_template_apply` with the property `template` matching the selected layout. 
In the console, you'll see an event like: 
`🔵 Tracked: editor_session_template_apply <template: contact-6>`

### Validate Editor Session
1. Create a new page from a selected layout using either of the paths mentioned above.
1. Perform some actions in the editor session like publishing, saving as a draft, or switching to the classic editor.
**Expect** to see the editor events `editor_session_start`, `editor_session_end`, and `editor_session_switch_editor` be populated with the property `template` matching the selected layout. 
In the console, you'll see an event like: 
`🔵 Tracked: editor_session_start <blog_type: wpcom, content_type: gutenberg, editor: gutenberg, has_unsupported_blocks: 0, post_type: page, session_id: 7C2AE7DA-9335-4FA8-AA0D-55D87628EDF5, startup_time_ms: 1707, template: contact-6, unsupported_blocks: (`
`🔵 Tracked: editor_session_end <blog_type: wpcom, content_type: gutenberg, editor: gutenberg, has_unsupported_blocks: 0, outcome: save, post_type: page, session_id: 7C2AE7DA-9335-4FA8-AA0D-55D87628EDF5, template: contact-6>`
`🔵 Tracked: editor_session_switch_editor <blog_type: wpcom, content_type: gutenberg, editor: classic, has_unsupported_blocks: 0, post_type: page, session_id: B6085F92-E875-4BD7-9FD3-22E8B5F87AC4, template: contact-6>`

### Create a Blank Page
1. Create a Blank Page
**Expect** The events mentioned in the last step to not have the template field populated. 

## Additional Context: 

🗒️ Instead of adding a third parameter to the list of completion block arguments, I did a bit of refactoring here to accept one argument which is the selected layout.

This is a continuation of the work done in https://github.com/wordpress-mobile/WordPress-iOS/pull/14987 and will be followed up by other tickets as part of [Modal Layout Picker Project](https://github.com/wordpress-mobile/gutenberg-mobile/issues?q=is%3Aissue+is%3Aopen+%5BModal+Layout+Picker%5D)

## PR submission checklist:
